File: QuestaoDois/EfeitoColateral.java
public class EfeitoColateral{

	private static int x = 0;

	public static void main (String[]args){
		System.out.print("1: ");
		efeito();
		System.out.print("2: ");
		efeito();
		System.out.print("3: ");
		efeito();
		System.out.print("4: ");
		efeito();

	}

	public static void efeito(){
		if(x < 2)
			System.out.println("Olá");
		x++;
	}
}

File: QuestaoDois/EfeitoColateral.c
#include <stdio.h>
int x = 0;

static void efeito()
{
    if(x < 2)
		printf("Olá");
	x++;    
}

int main()
{
    printf("1: ");
    efeito();
    printf("\n2: ");
    efeito();
    printf("\n3: ");
    efeito();
    printf("\n4: ");
    efeito();
    
    return 0;
}

